core/src/main/java/hudson/FilePath.java

```diff
@@ -1027,8 +1031,15 @@ private boolean installIfNecessaryFrom(@NonNull URL archive, @NonNull TaskListen
             }
 
             long sourceTimestamp = con.getLastModified();
+            String resultEtag = con.getHeaderField("ETag");
+            if (resultEtag != null && !resultEtag.isEmpty()) {
+                resultEtag = resultEtag.replace("\"", "");
```

Comment:
The tag can also be weak, in that case the etag sent is `W/"<value>" . So we need to remove a leading `W/` before removing the quotes

Reply:
I read the specification more carefully and realized that the original implementation was not handling the values as simply as it could.  Adjusted in df05da5bb5bfde52278bb25f27f77f894a1fd2eb .  Could you review again?